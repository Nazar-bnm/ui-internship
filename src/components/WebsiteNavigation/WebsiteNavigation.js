import React from 'react';

import { Button } from '@/shared';

import './WebsiteNavigation.scss';

const CN = 'website-navigation';

const WebsiteNavigation = ({ className }) => {
  const { pathname } = window.location;

  const goBack = () => {
    window.history.back();
  };

  return (
    <div className={`${CN}__wrapper content ${className}`}>
      <div className={`${CN}__pathname`}>
        Home
        {pathname}
      </div>
      <Button
        className={`${CN}__button`}
        onClick={goBack}
      >
        &#60; Return to the previous page
      </Button>
    </div>
  );
};

export default WebsiteNavigation;
