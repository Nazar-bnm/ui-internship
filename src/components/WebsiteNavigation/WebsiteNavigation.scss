@import '../../scss/global_styles';

.website-navigation {
  &__wrapper {
    @include flex;
    padding: 10px 0;
    border-top: 1px solid $silver-chalice;
    color: $silver-chalice;
    font-size: 0.7rem;
  }

  &__pathname {
    text-transform: capitalize;
  }

  &__button {
    background-color: transparent;
    border: none;
    font-size: 0.7rem;
    font-family: font(secondary);
    color: $silver-chalice;

    &:hover {
      text-decoration: underline;
    }
  }
}

.hidden {
  display: none;
}