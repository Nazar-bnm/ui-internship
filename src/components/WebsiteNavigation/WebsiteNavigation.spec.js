import React from 'react';
import { shallow } from 'enzyme';

import WebsiteNavigation from './WebsiteNavigation';

describe('<WebsiteNavigation />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<WebsiteNavigation />);
    expect(wrapper).toMatchSnapshot();
  });
});