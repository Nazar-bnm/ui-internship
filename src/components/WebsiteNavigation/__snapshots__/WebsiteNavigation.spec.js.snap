// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WebsiteNavigation /> should match snapshot 1`] = `
<div
  className="website-navigation__wrapper content undefined"
>
  <div
    className="website-navigation__pathname"
  >
    Home
    /
  </div>
  <Button
    className="website-navigation__button"
    icon={null}
    onClick={[Function]}
  >
    &lt; Return to the previous page
  </Button>
</div>
`;
